### PR TITLE
Gm/bloxssv 315 components redesign

### PR DESCRIPTION
**Pull request by guym-blox**

### CODE REVIEW

File: src/app/common/components/AppBar/AppBar.tsx [hunk 1]
```diff
@@ -42,17 +41,19 @@ const AppBarComponent = () => {
...
-                                }}
-                      className={classes.menuButton}>
-                      {button.text}
-                    </Typography>
+                    <Grid item xs={12}>
+                      <Typography
+                        key={index}
+                        onClick={() => {
+                                  if (button.active) { switchPage(button.link, button.newPage); }
+                              }}
```
Comment (meshin-blox) [inline src/app/common/components/AppBar/AppBar.tsx:52]: Reformat this section

File: src/app/common/components/AppBar/AppBar.tsx [hunk 2]
```diff
@@ -73,38 +74,51 @@ const AppBarComponent = () => {
...
+                                                  <Tooltip disableHoverListener={button.active} key={index} disableFocusListener disableTouchListener title="Comming soon...">
+                                                    <Typography onClick={() => { if (button.active) switchPage(button.link, button.newPage); }}
+                                                      variant={'subtitle1'}
+                                                      className={classes.button}>
+                                                      {button.text}
+                                                    </Typography>
+                                                  </Tooltip>
+                                                </Grid>
+                                              );
+                                          })}
```
Comment (meshin-blox) [inline src/app/common/components/AppBar/AppBar.tsx:103]: Reformat this, too many spaces, should be 2 more from parent

File: src/app/common/components/AppBar/AppBar.tsx [hunk 3]
```diff
@@ -73,38 +74,51 @@ const AppBarComponent = () => {
...
+                </Grid>
               </Grid>
-              <Grid item className={classes.menu} onClick={() => { handleClick(!applicationStore.shouldDisplayToolBar); }}>
-                {applicationStore.shouldDisplayToolBar ? <CloseIcon className={classes.menuIcon} /> : <MenuIcon className={classes.menuIcon} />}
+              <Grid item xs={8} md={4}>
+                <Grid container spacing={1} alignItems={'center'} justify={'flex-end'}>
+                  <Grid item>
+                    <ConnectWalletButton />
+                  </Grid>
+                  <Grid item className={classes.menu} onClick={() => { handleClick(!applicationStore.shouldDisplayToolBar); }}>
```
Comment (meshin-blox) [inline src/app/common/components/AppBar/AppBar.tsx:111]: Why to not check this condition in `handleClick` itself?

`!applicationStore.shouldDisplayToolBar`

File: src/app/common/components/DataSection/DataSection.tsx [hunk 4]
```diff
@@ -1,68 +1,34 @@
...
+            <Grid container spacing={0}>
+              {section.map((d: any, sectionIndex: number) => (
+                <Grid item className={classes.section} xs={12} key={sectionIndex}>
+                  <Grid container>
+                    <Grid item xs={6}>
+                      <Typography className={`${d.key === 'Total' ? classes.total : ''}`}>{d.key}</Typography>
+                    </Grid>
+                    <Grid item xs={6}>
+                      <Typography className={`${classes.dataValue} ${d.key === 'Total' ? classes.total : ''}`}>{
+                                        d.value}{d.strong && <strong>&nbsp;{d.strong}</strong>}
```
Comment (meshin-blox) [inline src/app/common/components/DataSection/DataSection.tsx:22]: Fix identation

File: src/app/common/components/DataSection/DataSection.tsx [hunk 5]
```diff
@@ -1,68 +1,34 @@
...
+                      </Typography>
+                    </Grid>
+                  </Grid>
+                </Grid>
+                    ))}
```
Comment (meshin-blox) [inline src/app/common/components/DataSection/DataSection.tsx:27]: Identation

File: src/app/common/components/Screen/Screen.styles.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import { makeStyles, Theme } from '@material-ui/core/styles';
+
+export const useStyles = makeStyles((theme: Theme) => ({
+    root: {
+    },
```
Comment (meshin-blox) [inline src/app/common/components/Screen/Screen.styles.ts:5]: Remove empty root,
also remove all commented style properties

File: src/app/common/components/Screen/Screen.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { observer } from 'mobx-react';
+import Grid from '@material-ui/core/Grid';
+import { useStyles } from '~app/common/components/Screen/Screen.styles';
+import Header from '~app/common/components/Header';
+import BackNavigation from '~app/common/components/BackNavigation';
```
Comment (meshin-blox) [inline src/app/common/components/Screen/Screen.tsx:6]: Rearrange imports

File: src/app/common/components/Screen/Screen.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    subTitle: string,
+    navigationText?: string
+    navigationOnClick?: any
+    navigationLink?: string
+    body?: any;
+    actionButton?: any;
+    align?: boolean;
+};
+
+const Screen = ({ icon, title, subTitle, navigationLink, navigationOnClick, navigationText, body, actionButton, align = false }: Params) => {
```
Comment (meshin-blox) [inline src/app/common/components/Screen/Screen.tsx:20]: Rename `Params` to `ScreenProps`

File: src/app/common/components/Screen/Screen.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+          <Header title={title} subtitle={subTitle} />
+        </Grid>
+        <Grid className={classes.body} item xs={12} md={12}>
+          { body }
+        </Grid>
+        {actionButton && (
+        <Grid item xs={12} md={12} className={classes.bottom}>
+          {actionButton}
+        </Grid>
+)}
```
Comment (meshin-blox) [inline src/app/common/components/Screen/Screen.tsx:43]: Fix identation

File: src/app/components/GenerateOperatorKeys/GenerateOperatorKeys.styles.ts [hunk 10]
```diff
@@ -12,28 +12,39 @@ export const useStyles = makeStyles((theme) => ({
...
+        marginTop: '12px',
         minHeight: '50px',
         lineHeight: '50px',
         backgroundColor: '#FDE6E5',
     },
     errorText: {
     },
     paddingTop: {
-        marginTop: '100px',
+        // marginTop: '100px',
```
Comment (meshin-blox) [inline src/app/components/GenerateOperatorKeys/GenerateOperatorKeys.styles.ts:36]: If style is not used remove it

File: src/app/components/GenerateOperatorKeys/GenerateOperatorKeys.tsx [hunk 11]
```diff
@@ -88,65 +84,69 @@ const GenerateOperatorKeys = () => {
...
+              {displayNameError.shouldDisplay && <Typography className={classes.textError}>{displayNameError.errorMessage}</Typography>}
+            </InputLabel>
+          </Grid>
+          <Grid item>
+            <InputLabel
+              title="Operator Key"
+              withHint
+              toolTipText={translations.OPERATOR.REGISTER.TOOL_TIP_KEY}
+              toolTipLink={config.links.TOOL_TIP_KEY_LINK}
+                  >
```
Comment (meshin-blox) [inline src/app/components/GenerateOperatorKeys/GenerateOperatorKeys.tsx:125]: Indentation

File: src/app/components/GenerateOperatorKeys/GenerateOperatorKeys.tsx [hunk 12]
```diff
@@ -88,65 +84,69 @@ const GenerateOperatorKeys = () => {
...
+              <TextInput type="text"
+                data-testid="new-operator-key"
+                className={publicKeyError.shouldDisplay ? classes.inputError : ''}
+                onChange={(event) => { onInputChange('publicKey', event.target.value); }}
+                onBlur={(event) => { validatePublicKeyInput(event.target.value, setPublicKeyError); }}
+                value={inputsData.publicKey}
+                    />
```
Comment (meshin-blox) [inline src/app/components/GenerateOperatorKeys/GenerateOperatorKeys.tsx:132]: Indentation

File: src/app/components/GenerateOperatorKeys/GenerateOperatorKeys.tsx [hunk 13]
```diff
@@ -88,65 +84,69 @@ const GenerateOperatorKeys = () => {
...
-      </Grid>
-    </Paper>
+      )}
+      actionButton={(
+        <CTAButton
+          testId="register-operator-button"
+          disable={!registerButtonEnabled}
+          onClick={onRegisterClick}
+          text={'Next'}
+          />
```
Comment (meshin-blox) [inline src/app/components/GenerateOperatorKeys/GenerateOperatorKeys.tsx:147]: Indentation

File: src/app/components/ImportValidatorConfirmation/ImportValidatorConfirmation.tsx [hunk 14]
```diff
@@ -48,62 +45,43 @@ const ImportValidatorConfirmation = () => {
...
+        [
+            { key: 'Est. Transaction Cost', value: '' },
+            { key: 'Network fee', value: 'FREE', strong: '$0.00' },
+            { key: 'Transaction fee', value: 'FREE', strong: `$${normalizeNumber(contractValidator.dollarEstimationGas)}` },
+            { key: 'Total', value: '', strong: `${normalizeNumber(contractValidator.dollarEstimationGas)}` },
+       ], 
+    ];
+
+    contractOperator.operators.forEach((operator: IOperator, index: number) => {
+          if (operator.selected) data[0].push({ key: `${index + 1}. ${operator.name}`, value: longStringShorten(sha256(walletStore.decodeOperatorKey(operator.pubkey)), 4) });
```
Comment (meshin-blox) [inline src/app/components/ImportValidatorConfirmation/ImportValidatorConfirmation.tsx:61]: Too long line

File: src/app/components/RegisterOperatorHome/RegisterOperatorHome.tsx [hunk 15]
```diff
@@ -1,18 +1,15 @@
 import React from 'react';
 import { observer } from 'mobx-react';
-import Grid from '@material-ui/core/Grid';
-import Paper from '@material-ui/core/Paper';
 import { useHistory } from 'react-router-dom';
-import { useStores } from '~app/hooks/useStores';
-import Header from '~app/common/components/Header';
-import Typography from '@material-ui/core/Typography';
+import Grid from '@material-ui/core/Grid';
```
Comment (meshin-blox) [inline src/app/components/RegisterOperatorHome/RegisterOperatorHome.tsx:4]: Reorder imports

File: src/app/components/RegisterValidatorHome/RegisterValidatorHome.tsx [hunk 16]
```diff
@@ -29,39 +27,27 @@ const RegisterValidatorHome = () => {
...
+    <Screen
+      navigationText={translations.HOME.TITLE}
+      navigationLink={config.routes.HOME}
+      title={translations.VALIDATOR.HOME.TITLE}
+      subTitle={translations.VALIDATOR.HOME.DESCRIPTION}
+      body={(
+        <Grid container wrap="nowrap" spacing={3} className={classes.columnGridContainer}>
+          <Grid item xs={12} md={12} zeroMinWidth className={classes.columnGridContainer}>
+            <ConditionalLink to={config.routes.VALIDATOR.CREATE} condition onClick={() => redirectTo(config.routes.VALIDATOR.CREATE)}>
+              <LinkButton text1={'Create Validator'} text2={''} />
```
Comment (meshin-blox) [inline src/app/components/RegisterValidatorHome/RegisterValidatorHome.tsx:39]: What is text1 and text2, can we name these props with better sense?
Like, `primaryLabel`, `secondaryLabel`

File: src/app/components/RegisterValidatorHome/RegisterValidatorHome.tsx [hunk 17]
```diff
@@ -29,39 +27,27 @@ const RegisterValidatorHome = () => {
...
-                <Grid item md={12} xs={12} className={classes.bigSquareButtonGrid}>
-                  <img src="/images/etherium.png" alt="Import Validator" className={classes.bigSquareButtonIcon} />
-                  <Typography noWrap variant="h6" className={classes.guideStepText}>Import Validator</Typography>
-                </Grid>
-              </Grid>
-            </Paper>
-          </ConditionalLink>
+          <Grid item xs={12} md={12} zeroMinWidth className={classes.columnGridContainer}>
+            <ConditionalLink to={config.routes.VALIDATOR.IMPORT} condition={walletStore.connected} onClick={() => redirectTo(config.routes.VALIDATOR.IMPORT)}>
+              <LinkButton text1={'Import Validator'} text2={''} />
```
Comment (meshin-blox) [inline src/app/components/RegisterValidatorHome/RegisterValidatorHome.tsx:45]: The same naming issue

File: src/app/components/RegisterValidatorHome/components/CreateValidator/CreateValidator.styles.ts [hunk 18]
```diff
@@ -1,123 +1,27 @@
...
+      fontStyle: 'normal',
+      fontFamily: 'Inter, sans-serif',
+      fontWeight: 500,
     },
     guideStepsContainerPaper: {
         cursor: 'pointer',
         border: '1px solid black',
         marginTop: '60px',
-        margin: `${theme.spacing(1)}px auto`,
+        // margin: `${theme.spacing(1)}px auto`,
```
Comment (meshin-blox) [inline src/app/components/RegisterValidatorHome/components/CreateValidator/CreateValidator.styles.ts:19]: if not using, remove

File: src/app/components/RegisterValidatorHome/components/CreateValidator/CreateValidator.styles.ts [hunk 19]
```diff
@@ -129,11 +33,19 @@ export const useStyles = makeStyles((theme) => ({
...
         float: 'right',
+        width: '125px',
+        /*  display: 'block',
+          float: 'right',
+          position: 'relative',
+          [theme.breakpoints.down('xl')]: {
+              top: '60px',
+          }, */
+        // right: '1px',
+        // bottom: '1px',
```
Comment (meshin-blox) [inline src/app/components/RegisterValidatorHome/components/CreateValidator/CreateValidator.styles.ts:49]: The same about not used styles

File: src/app/components/RegisterValidatorHome/components/CreateValidator/CreateValidator.tsx [hunk 20]
```diff
@@ -1,14 +1,12 @@
...
-import Paper from '@material-ui/core/Paper';
 import Typography from '@material-ui/core/Typography';
-import Header from '~app/common/components/Header';
 import config, { translations } from '~app/common/config';
-import BackNavigation from '~app/common/components/BackNavigation';
-import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos';
+import Screen from '~app/common/components/Screen/Screen';
 import ConditionalLink from '~app/common/components/ConditionalLink';
 import { useStyles } from '~app/components/RegisterValidatorHome/components/CreateValidator/CreateValidator.styles';
+import LinkButton from '~app/common/components/LinkButton/LinkButton';
```
Comment (meshin-blox) [inline src/app/components/RegisterValidatorHome/components/CreateValidator/CreateValidator.tsx:9]: Rearrange imports

File: src/app/components/RegisterValidatorHome/components/CreateValidator/CreateValidator.tsx [hunk 21]
```diff
@@ -18,40 +16,37 @@ const CreateValidator = () => {
...
+                return (
+                  <Grid key={text} item>
+                    <Typography className={classes.bodyText} variant="subtitle1">
+                      {text}
+                    </Typography>
+                  </Grid>
+                );
+              })}
+            </Grid>
+          </Grid>
```
Comment (meshin-blox) [inline src/app/components/RegisterValidatorHome/components/CreateValidator/CreateValidator.tsx:38]: These two greeds are on the same indentation level

File: src/app/components/RegisterValidatorHome/components/FileApproval/FileApproval.tsx [hunk 22]
```diff
@@ -57,54 +55,66 @@ const EnterValidatorPrivateKey = () => {
     validatorStore.setValidatorPrivateKeyFile(false);
     history.push(config.routes.VALIDATOR.IMPORT);
   };
+  
+  const serializeFileName = (fileName: string) => {
+    if (fileName.length > 34 && isMobile) {
+      return `${fileName.slice(0, 12)}...${fileName.slice(fileName.length - 14, fileName.length)}`;
+    } 
+      return fileName;
```
Comment (meshin-blox) [inline src/app/components/RegisterValidatorHome/components/FileApproval/FileApproval.tsx:63]: Indentation

File: src/app/components/RegisterValidatorHome/components/FileApproval/FileApproval.tsx [hunk 23]
```diff
@@ -57,54 +55,66 @@ const EnterValidatorPrivateKey = () => {
...
+                value={validatorStore.password}
+                onChange={(event: any) => validatorStore.setPassword(event.target.value)}
+                    />
+            </InputLabel>
+            <Grid className={classes.paddingTop} item xs={12}>
+              {showErrorMessage && (
+                <div className={classes.errorDiv}>
+                  {errorMessage}
+                </div>
+                    )}
```
Comment (meshin-blox) [inline src/app/components/RegisterValidatorHome/components/FileApproval/FileApproval.tsx:104]: Indentation

File: src/app/components/RegisterValidatorHome/components/SelectOperators/SelectOperators.tsx [hunk 24]
```diff
@@ -1,26 +1,23 @@
...
-import Header from '~app/common/components/Header';
 import CTAButton from '~app/common/components/CTAButton';
 import config, { translations } from '~app/common/config';
-import OperatorSelector from './components/OperatorSelector';
+import Screen from '~app/common/components/Screen/Screen';
 import ApplicationStore from '~app/common/stores/Application.store';
-import HistoryBackNavigation from '~app/common/components/HistoryBackNavigation';
-import ContractOperator, { IOperator } from '~app/common/stores/contract/ContractOperator.store';
 import { useStyles } from '~app/components/GenerateOperatorKeys/GenerateOperatorKeys.styles';
+import ContractOperator, { IOperator } from '~app/common/stores/contract/ContractOperator.store';
```
Comment (meshin-blox) [inline src/app/components/RegisterValidatorHome/components/SelectOperators/SelectOperators.tsx:13]: Rearrange all imports

File: src/app/components/RegisterValidatorHome/components/SelectOperators/SelectOperators.tsx [hunk 25]
```diff
@@ -50,40 +47,43 @@ const SelectOperators = () => {
...
+                disabled={!contractOperator.operators.length}
+                variant="contained"
+                color="primary"
+                style={{ width: '100%' }}
+                onClick={contractOperator.autoSelectOperators}
+                      >
+                <AutorenewIcon />
+                        &nbsp;Auto-select
+              </Button>
+                  ) : ''}
```
Comment (meshin-blox) [inline src/app/components/RegisterValidatorHome/components/SelectOperators/SelectOperators.tsx:70]: Indentation

File: src/app/components/RegisterValidatorHome/components/SlashingWarning/SlashingWarning.tsx [hunk 26]
```diff
@@ -1,15 +1,13 @@
 import React, { useEffect, useState } from 'react';
 import { observer } from 'mobx-react';
 import Grid from '@material-ui/core/Grid';
-import Paper from '@material-ui/core/Paper';
 import Checkbox from '@material-ui/core/Checkbox';
 import Typography from '@material-ui/core/Typography';
 import FormControlLabel from '@material-ui/core/FormControlLabel';
```
Comment (meshin-blox) [inline src/app/components/RegisterValidatorHome/components/SlashingWarning/SlashingWarning.tsx:6]: Imports arrange

File: src/app/components/RegisterValidatorHome/components/SlashingWarning/SlashingWarning.tsx [hunk 27]
```diff
@@ -53,54 +49,60 @@ const SlashingWarning = () => {
...
                 checked={userAgreed}
-                onChange={(event) => {
-                  setUserAgreed(event.target.checked);
-                }}
+                onChange={(event) => { setUserAgreed(event.target.checked); }}
                 color="primary"
-              />
-            )}
+                      />
+                  )}
```
Comment (meshin-blox) [inline src/app/components/RegisterValidatorHome/components/SlashingWarning/SlashingWarning.tsx:87]: Indentation

File: src/app/components/RegisterValidatorHome/components/SlashingWarning/SlashingWarning.tsx [hunk 28]
```diff
@@ -53,54 +49,60 @@ const SlashingWarning = () => {
...
+              <Typography className={classes.checkboxText}>
+                I understand that running my validator simultaneously in multiple setups will cause slashing
+                to my validator.
               </Typography>
-            )}
-            style={{ marginTop: 150 }}
-          />
-
+                  )}
+              />
```
Comment (meshin-blox) [inline src/app/components/RegisterValidatorHome/components/SlashingWarning/SlashingWarning.tsx:94]: Indentation

File: src/app/components/SuccessScreen/SuccessScreen.tsx [hunk 29]
```diff
@@ -1,16 +1,15 @@
...
 import useUserFlow from '~app/hooks/useUserFlow';
-import { translations } from '~app/common/config';
-import Header from '~app/common/components/Header';
+import config, { translations } from '~app/common/config';
 import { useStyles } from '~app/components/SuccessScreen/SuccessScreen.styles';
 import ContractOperator from '~app/common/stores/contract/ContractOperator.store';
 import ContractValidator from '~app/common/stores/contract/ContractValidator.store';
+import Screen from '~app/common/components/Screen/Screen';
+import LinkButton from '~app/common/components/LinkButton/LinkButton';
+import ConditionalLink from '~app/common/components/ConditionalLink';
```
Comment (meshin-blox) [inline src/app/components/SuccessScreen/SuccessScreen.tsx:12]: Rearrange imports

File: src/app/components/SuccessScreen/SuccessScreen.tsx [hunk 30]
```diff
@@ -23,39 +22,36 @@ const SuccessScreen = () => {
...
+        <Grid className={classes.gridContainer} container spacing={5}>
+          <Grid item xs>
+            <ConditionalLink to={config.routes.OPERATOR.GENERATE_KEYS} condition={false} onClick={(() => { })}>
+              <LinkButton text1={monitorHeader} text2={monitorText} />
+            </ConditionalLink>
           </Grid>
-        </Paper>
-      </Grid>
-    </Paper>
+        </Grid>
```
Comment (meshin-blox) [inline src/app/components/SuccessScreen/SuccessScreen.tsx:52]: Indentation

File: src/app/components/WalletPopUp/WalletPopUp.styles.ts [hunk 31]
```diff
@@ -1,33 +1,36 @@
...
     },
-    gridContainerAddress: {
-        maxWidth: '100%',
-        borderRadius: '10px',
-        margin: 'auto',
-        backgroundColor: '#F2F2F2',
-    },
     dialogWrapper: {
-        border: '1px solid black',
+        // border: '1px solid black',
```
Comment (meshin-blox) [inline src/app/components/WalletPopUp/WalletPopUp.styles.ts:33]: Remove

File: src/app/components/WalletPopUp/WalletPopUp.tsx [hunk 32]
```diff
@@ -1,23 +1,21 @@
 import React from 'react';
 import { observer } from 'mobx-react';
+import { isMobile } from 'react-device-detect';
```
Comment (meshin-blox) [inline src/app/components/WalletPopUp/WalletPopUp.tsx:3]: Rearrange

**Review by meshin-blox — CHANGES_REQUESTED** [review]

File: src/app/common/components/AppBar/AppBar.tsx (hunk 3 above)
Comment (guym-blox) [reply src/app/common/components/AppBar/AppBar.tsx:111]: because the function not related to view rendering

**Pull request merged**